documentation/High-level-overview.md

```diff
@@ -0,0 +1,166 @@
+
+## Tasks
+
+## Diagnosability / Loggers
+Diagnosability within MSBuild went through some changes. Before we had a debugger, where you could step through the XML during the build and debug. This was discarded in favor of a log focused approach, where MSBuild has a more robust logging system that contains more data to identify what is happening during a build.
+
+Beyond loggers, we have some ETW (Event Tracing for Windows) events which can also be identified through loggers.
+
+### General Loggers
+Logging within MSBuild consists of various integrated and pluggable loggers. Integrated loggers generally processes code structure events, such as communication between nodes during build, or data for BuildCheck analyzers to run properly. Built-in loggers include the Binary Logger, Console / Terminal logger, and a Text Log. Pluggable loggers are third party loggers that can receive events through the MSBuild API, or the .NET event handlers.
+
+Pluggable loggers are added through DLLs, and MSBuild engine identifies them at the beginning of the build. Because of this, build logic is not able to manipulate loggers.
+
+### Binary logger
+The Binary Logger, also called BinLog, is a structured log that contains all the events within a build. It achieves that through its implementation focused on reading events from the build and serializing those in an structured form. To read a BinLog the BinLog reader can be used, but it is not officially supported by the MSBuild team.
+It is one of the best tools for debugging MSBuild.
+
+
+## Project result cache plugin
+This Project Cache differs from the previous one because it is separate from the main MSBuild code and used mainly for distributed builds. It functions as a middle layer between the scheduler and the Project Result Cache. So, when the scheduler requests a result for a target or project, the plugin responds first to check all the different distributed nodes for the result. To accomplish this, it adds profiles on disk based on hashes of the project or task ID / name. When the plugin cache gets a hit on an input, it downloads and copies the file results to the right place, deserializes the resulting payload and provides it to the local engine to continue execution.
+
+For more in depth information visit [the spec](../documentation/specs/project-cache.md).
+
+## BuildCheck
+
+## Resolvers
+There are a few elements within the MSBuild XML that indicate that a call to the .NET SDK is necessary. Some examples include:
+ - `<Project Sdk="Microsoft.NET.Sdk">`, where you can also define the SDK version
+ - `<Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />`, for explicit imports.
+ - `<Sdk Name="My.Build.Sdk" Version="1.0.0" />`, another explicit import of the SDK.
```

Comment:
Nothing else has this level of detail. Simplify to "When an MSBuild SDK is referenced, it's run through the resolvers"?

Reply:
The doc is called High Level Overview!? So, this level of detail should be expected, right?

Reply:
I think the description is a good level of detail, I just removed the examples as something like this is not present in the rest of the document.